src/framework/audio/internal/worker/mixer.cpp

```diff
@@ -179,14 +179,18 @@ samples_t Mixer::process(float* outBuffer, samples_t samplesPerChannel)
 
     prepareAuxBuffers(outBufferSize);
 
-    for (size_t trackIdx = 0; trackIdx < futureList.size(); ++trackIdx) {
+    size_t trackIdx = 0;
```

Comment:
Replace 

`std::vector<std::future<std::vector<float> > > futureList` 

with 

`std::map<TrackId, std::future<std::vector<float>> futures`

After that, we can iterate over this map:

```
for (auto& pair : futures) {
    const std::vector<float>& trackBuffer = pair.second.get();

    ...

    const AuxSendsParams& auxSends = m_trackChannels.at(pair.first)->outputParams().auxSends;

    ...
}
```

It's a slightly better looking solution